index.d.ts

```diff
@@ -1,32 +1,31 @@
 // TypeScript Version: 2.2
 import * as React from 'react';
 
-declare namespace DayPicker {
```

Comment:
in general this is a good change because it retains backwards compatibility for the top-level symbol exports.

_however_, you need to make the export include the `DayPicker` class, and the only way to do that is with `export = DayPicker` and declaration merging. so you're going to need to keep this `namespace` as it is. I recommend putting the `DayPicker` class in new file and importing it as `import { DayPicker } from "./DayPicker"`. when you import it here, it will merge with this namespace and you can `export = DayPicker` at the bottom. the diff in this file should be fairly small (not 100s of lines).

Reply:
@adidahiya Was I clear enough explaining the problem I am encountering following your recommendation? If time permits, I would be more than happy to have your input so I can move forward with this.

Reply:
I made a new PR to provide the same funtionality as this one: https://github.com/gpbl/react-day-picker/pull/487

My declaration merging idea didn't work out, but you should be able to use those types and import like this:

```ts
import * as DayPicker from "react-day-picker";
import * as DayPickerInput from "react-day-picker/DayPickerInput";
```